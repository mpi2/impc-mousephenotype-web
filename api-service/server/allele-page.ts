import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { AlleleSummary } from "@/models";
const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchAlleleSummary(
  mgiGeneAccessionId: string,
  alleleName: string,
): Promise<AlleleSummary> {
  const endpointURL = `http://impc-allele-service.${KUBERNETES_NAMESPACE}:8080/v1/alleles?mgiGeneAccessionId=${mgiGeneAccessionId}&alleleName=${alleleName}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<AlleleSummary>(
        `/api/v1/alleles/${mgiGeneAccessionId}/${alleleName}`,
      )
    : fetchURL<AlleleSummary>(endpointURL));
}
