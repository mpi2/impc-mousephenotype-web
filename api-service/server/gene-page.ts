import { GeneOrder, GenePhenotypeHits, GeneSummary } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { processGeneOrderResponse } from "@/hooks/gene-order.query";
import { processGenePhenotypeHitsResponse } from "@/hooks/significant-phenotypes.query";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchGeneSummary(
  mgiGeneAccessionId: string,
): Promise<GeneSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/gene_summary?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<GeneSummary>(
        `/api/v1/genes/${mgiGeneAccessionId}/summary`,
      )
    : fetchURL<GeneSummary>(endpointURL));
}

export async function fetchGenePhenotypeHits(
  mgiGeneAccessionId: string,
): Promise<Array<GenePhenotypeHits>> {
  const endpointURL = `http://impc-phenotype-hits-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotypehits?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GenePhenotypeHits>>(
        `/api/v1/genes/${mgiGeneAccessionId}/phenotype-hits`,
      )
    : fetchURL<Array<GenePhenotypeHits>>(endpointURL));
  return processGenePhenotypeHitsResponse(data);
}

export async function fetchGeneOrderData(
  mgiGeneAccessionId: string,
): Promise<Array<GeneOrder>> {
  const endpointURL = `http://impc-order-service.${KUBERNETES_NAMESPACE}:8080/v1/orders?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GeneOrder>>(
        `/api/v1/genes/${mgiGeneAccessionId}/order`,
      )
    : fetchURL<Array<GeneOrder>>(endpointURL));
  return processGeneOrderResponse(data);
}
