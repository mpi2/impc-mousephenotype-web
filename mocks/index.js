export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

