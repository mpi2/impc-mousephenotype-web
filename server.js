const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});