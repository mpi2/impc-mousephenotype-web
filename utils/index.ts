import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
