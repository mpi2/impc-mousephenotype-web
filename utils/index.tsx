import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
