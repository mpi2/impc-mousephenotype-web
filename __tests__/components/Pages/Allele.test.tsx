import AllelePage from "@/app/alleles/[pid]/[...alleleSymbol]/allele-page";
import "@testing-library/jest-dom";
import { API_URL, renderWithClient } from "../../utils";
import { testServer } from "../../../mocks/server";
import { rest } from "msw";
import dbn1Tm1aEsCellData from "../../../mocks/data/tests/dbn1-tm1a-es-cell.json";
import dbn1Tm1aMiceData from "../../../mocks/data/tests/dbn1-tm1a-mice.json";
import dbn1Tm1aTvpData from "../../../mocks/data/tests/dbn1-tm1a-tvp.json";
import dbn1Em1MiceData from "../../../mocks/data/tests/dbn1-em1-mice-data.json";
import dbn1Em1CrisprData from "../../../mocks/data/tests/dbn1-em1-crispr-data.json";
import { screen } from "@testing-library/react";

const alleleData = {
  alleleDescription:
    "KO first allele (reporter-tagged insertion with conditional potential)",
  alleleName: "tm1a(KOMP)Wtsi",
  doesCrisprProductsExist: false,
  doesEsCellProductsExist: true,
  doesIntermediateVectorProductsExist: false,
  doesMiceProductsExist: true,
  doesTargetingVectorProductsExist: true,
  emsembleUrl:
    "https://www.ensembl.org/Mus_musculus/Location/View?g=ENSMUSG00000034675",
  geneSymbol: "Dbn1",
  id: "67176b9293943bea03ef2819",
  mgiGeneAccessionId: "MGI:1931838",
};

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useSearchParams: jest.fn().mockImplementation(() => new URLSearchParams()),
  usePathname: jest.fn(),
  useParams: jest.fn().mockImplementation(() => ({
    pid: alleleData.mgiGeneAccessionId,
    alleleSymbol: alleleData.alleleName,
  })),
}));

jest.mock("@/components/GenomeBrowser/GenomeBrowser", () => {
  const GenomeBrowserMock = () => <div>Genome Browser mock</div>;
  return GenomeBrowserMock;
});

describe("Allele page", () => {
  it("renders properly", async () => {
    testServer.use(
      rest.get(
        `${API_URL}/api/v1/alleles/es_cell/get_by_mgi_and_allele_name/MGI:1931838/tm1a\\(KOMP\\)Wtsi`,
        (_, res, ctx) => {
          return res(ctx.json(dbn1Tm1aEsCellData));
        },
      ),
      rest.get(
        `${API_URL}/api/v1/alleles/mice/get_by_mgi_and_allele_name/MGI:1931838/tm1a\\(KOMP\\)Wtsi`,
        (_, res, ctx) => {
          return res(ctx.json(dbn1Tm1aMiceData));
        },
      ),
      rest.get(
        `${API_URL}/api/v1/alleles/tvp/get_by_mgi_and_allele_name/MGI:1931838/tm1a\\(KOMP\\)Wtsi`,
        (_, res, ctx) => {
          return res(ctx.json(dbn1Tm1aTvpData));
        },
      ),
    );
    const { container } = renderWithClient(
      <AllelePage alleleSymbol="tm1a(KOMP)Wtsi" alleleData={alleleData} />,
    );
    expect(await screen.findByTestId("es-cell-section")).toBeInTheDocument();
    expect(await screen.findByTestId("mice-section")).toBeInTheDocument();
    expect(await screen.findByTestId("tvp-section")).toBeInTheDocument();
    expect(screen.queryByTestId("crispr-section")).toBeNull();
    expect(container).toMatchSnapshot();
  });
  it("renders properly CRISPR section", async () => {
    testServer.use(
      rest.get(
        `${API_URL}/api/v1/alleles/mice/get_by_mgi_and_allele_name/MGI:1931838/em1\\(IMPC\\)Bay`,
        (req, res, ctx) => {
          return res(ctx.json(dbn1Em1MiceData));
        },
      ),
      rest.get(
        `${API_URL}/api/v1/alleles/crispr/get_by_mgi_and_allele_superscript/MGI:1931838/em1\\(IMPC\\)Bay`,
        (req, res, ctx) => {
          return res(ctx.json(dbn1Em1CrisprData));
        },
      ),
    );
    const { container } = renderWithClient(
      <AllelePage
        alleleSymbol="em1(IMPC)Bay"
        alleleData={{
          ...alleleData,
          alleleName: "em1(IMPC)Bay",
          alleleDescription: "Exon Deletion",
          doesCrisprProductsExist: true,
          doesEsCellProductsExist: false,
          doesIntermediateVectorProductsExist: false,
          doesMiceProductsExist: true,
          doesTargetingVectorProductsExist: false,
        }}
      />,
    );
    expect(await screen.findByTestId("mice-section")).toBeInTheDocument();
    expect(await screen.findByTestId("crispr-section")).toBeInTheDocument();
    expect(screen.queryByTestId("es-cell-section")).toBeNull();
    expect(screen.queryByTestId("tvp-section")).toBeNull();
    expect(container).toMatchSnapshot();
  });
});
