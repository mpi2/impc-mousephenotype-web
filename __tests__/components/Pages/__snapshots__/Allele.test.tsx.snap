// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllelePage renders properly 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="pb-4 pt-5 container container"
    >
      <div
        class="col-12 col-md-8 ps-4 pe-4"
      >
        <div
          class="tabs"
        >
          <a
            class="tab__active"
            href="/search"
          >
            Genes
          </a>
          <a
            class="tab"
            href="/search?type=pheno"
          >
            Phenotypes
          </a>
          <a
            class="tab"
            href="https://dev.mousephenotype.org/?s="
          >
            Help, news, blog
          </a>
        </div>
        <div
          class="inputCont"
        >
          <input
            class="input"
            placeholder="Search for a gene..."
            title="main search box"
            type="text"
            value=""
          />
          <button
            aria-describedby="svg-inline--fa-title-search-icon"
            class="searchBtn"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-search-icon"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-search-icon"
              >
                Search button
              </title>
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page container"
  >
    <div
      class="card"
    >
      <div
        class="subheading"
      >
        <span
          class="subheadingSection primary"
        >
          <a
            class="mb-3"
            href="/genes/MGI:1931838"
            style="text-transform: none; font-weight: normal; letter-spacing: normal; font-size: 1.15rem;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-left "
              data-icon="arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
            Â  Go Back to 
            <i>
              Dbn1
            </i>
          </a>
        </span>
      </div>
      <p
        class="subheading mt-2"
      >
        ALLELE
      </p>
      <h1
        class="mb-2 mt-2"
      >
        <i>
          Dbn1
          <sup>
            tm1a(KOMP)Wtsi
          </sup>
        </i>
      </h1>
      <p
        class="mb-4 grey"
      >
        KO first allele (reporter-tagged insertion with conditional potential)
      </p>
      <div
        style="display: flex; flex-wrap: wrap;"
      >
        <div
          class="dataCollection"
        >
          Mice
          <p
            class="mt-2"
          >
            <a
              class="btn impc-primary-button"
              href="#mice"
              style="min-width: 120px; cursor: pointer;"
            >
              <span>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-cart-shopping "
                  data-icon="cart-shopping"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                    fill="currentColor"
                  />
                </svg>
                Order
              </span>
            </a>
          </p>
        </div>
        <div
          class="dataCollection"
        >
          Targeted ES cells
          <p
            class="mt-2"
          >
            <a
              class="btn impc-primary-button"
              href="#esCell"
              style="min-width: 120px; cursor: pointer;"
            >
              <span>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-cart-shopping "
                  data-icon="cart-shopping"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                    fill="currentColor"
                  />
                </svg>
                Order
              </span>
            </a>
          </p>
        </div>
        <div
          class="dataCollection"
        >
          Targeting vectors
          <p
            class="mt-2"
          >
            <a
              class="btn impc-primary-button"
              href="#targetingVector"
              style="min-width: 120px; cursor: pointer;"
            >
              <span>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-cart-shopping "
                  data-icon="cart-shopping"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                    fill="currentColor"
                  />
                </svg>
                Order
              </span>
            </a>
          </p>
        </div>
        <div
          class="dataCollectionInactive"
        >
          Intermediate vectors
          <p
            class="mt-2"
          >
            <a
              class="btn btn-grey impc-base-button"
              href="#intermediateVector"
              style="min-width: 120px; cursor: initial;"
            >
              Not available
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="card"
      id="mice"
    >
      <h2>
        Mice
      </h2>
      <div
        class="buttonsWrapper top "
        data-testid="top-controls-wrapper"
      />
      <div
        class="tableWrapper"
      >
        <table
          class="table striped table table-bordered"
        >
          <thead>
            <tr>
              <th
                rowspan="2"
                width="25%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Colony Name
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Genetic Background
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Production Centre
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  QC Data
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  ES Cell/Parent Mouse Colony
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Order / Contact
                   
                </button>
              </th>
            </tr>
            <tr />
          </thead>
          <tbody />
        </table>
      </div>
      <div
        class="buttonsWrapper withControls"
      >
        <div
          class="additionalWrapper bottomControls"
        >
          <div
            class="grey"
            style="display: flex; gap: 0.5rem; align-items: center;"
          >
            Download data as:
            Â 
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               TSV
            </button>
            Â 
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               XLS
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
      id="esCell"
    >
      <h2>
        ES Cells
      </h2>
      <div
        class="buttonsWrapper top "
        data-testid="top-controls-wrapper"
      />
      <div
        class="tableWrapper"
      >
        <table
          class="table striped table table-bordered"
        >
          <thead>
            <tr>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  ES Cell Clone
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  ES Cell strain
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Parental Cell Line
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  IKMC Project
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  QC Data
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Targeting Vector
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Order / Contact
                   
                </button>
              </th>
            </tr>
            <tr />
          </thead>
          <tbody />
        </table>
      </div>
      <div
        class="buttonsWrapper withControls"
      >
        <div
          class="additionalWrapper bottomControls"
        >
          <div
            class="grey"
            style="display: flex; gap: 0.5rem; align-items: center;"
          >
            Download data as:
            Â 
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               TSV
            </button>
            Â 
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               XLS
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
      id="targetingVector"
    >
      <h2>
        Targeting vectors
      </h2>
      <div
        class="buttonsWrapper top "
        data-testid="top-controls-wrapper"
      />
      <div
        class="tableWrapper"
      >
        <table
          class="table striped table table-bordered"
        >
          <thead>
            <tr>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Design Oligos
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Targeting Vector
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Cassette
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Backbone
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="12.5%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  IKMC Project
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="12.5%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Genbank File
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="12.5%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Vector Map
                   
                </button>
              </th>
              <th
                rowspan="2"
                width="12.5%"
              >
                <button
                  class="inlineButton headerButton disabled"
                  style="font-weight: inherit;"
                >
                  Order
                   
                </button>
              </th>
            </tr>
            <tr />
          </thead>
          <tbody />
        </table>
      </div>
      <div
        class="buttonsWrapper withControls"
      >
        <div
          class="additionalWrapper bottomControls"
        >
          <div
            class="grey"
            style="display: flex; gap: 0.5rem; align-items: center;"
          >
            Download data as:
            Â 
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               TSV
            </button>
            Â 
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               XLS
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      Genome Browser mock
    </div>
    <div
      class="card"
    >
      <a
        class="primary link"
        href="/genes/MGI:1931838#order"
      >
        See all alleles for the gene
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right-long "
          data-icon="arrow-right-long"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
  <script
    type="application/ld+json"
  >
    {"@type":"Dataset","@context":"http://schema.org","name":"Mouse allele Dbn1&lt;tm1a(KOMP)Wtsi&gt;","description":"Discover mouse allele tm1a(KOMP)Wtsi of Dbn1 gene, view all available products and tissues with their detailed information.","creator":[{"@type":"Organization","name":"International Mouse Phenotyping Consortium"}],"citation":"https://doi.org/10.1093/nar/gkac972","isAccessibleForFree":true,"url":"https://dev.mousephenotype.org/data/alleles/MGI:1931838/tm1a(KOMP)Wtsi","license":"https://creativecommons.org/licenses/by/4.0/"}
  </script>
</div>
`;
